aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -708,4 +768,36 @@ public void onServiceDisconnected(final ComponentName name) {
             }
         };
     }
+
+    // Inspects context to determine whether HostedUIRedirectActivity is declared in
+    // customer's AndroidManifest.xml.
+    private boolean isRedirectActivityDeclared() {
+        // If the activity was found at least once, then don't bother searching again.
+        if (isRedirectActivityDeclared) {
+            return true;
+        }
+        if (context == null) {
+            Log.w(TAG, "Context is null. Failed to inspect packages.");
+            return false;
+        }
+        try {
+            List<PackageInfo> packages = context.getPackageManager()
+                    .getInstalledPackages(PackageManager.GET_ACTIVITIES);
+            for (PackageInfo packageInfo : packages) {
+                if (packageInfo.activities == null) {
+                    continue;
+                }
+                for (ActivityInfo activityInfo : packageInfo.activities) {
+                    if (activityInfo.name.contains(REDIRECT_ACTIVITY_NAME)) {
+                        isRedirectActivityDeclared = true;
+                        return true;
+                    }
+                }
+            }
+            Log.w(TAG, REDIRECT_ACTIVITY_NAME + " is not declared in AndroidManifest.");
+        } catch (Exception error) {
+            Log.w(TAG, "Failed to inspect packages.");
+        }
+        return false;
+    }
```

Comment:
This is to make sure that apps that:
- use the updated library
- did not update the code to use new `HostedUIRedirectActivity`
 
will not be held hostage by the signout-wait-latch that will never count down due to lack of callback event. We will inspect their android manifest to make sure that they followed the newest documentation to use `HostedUIRedirectActivity` before enabling the latch that fixes the race condition.